src/pages/studyView/virtualStudy/VirtualStudy.tsx

```diff
@@ -107,17 +108,7 @@ export default class VirtualStudy extends React.Component<
         super(props);
         makeObservable(this);
         this.name = props.name || '';
-        this.description =
-            props.description ||
-            getVirtualStudyDescription(
-                this.props.description,
-                this.props.studyWithSamples,
-                this.props.filter,
-                this.attributeNamesSet,
-                this.props.molecularProfileNameSet,
-                this.props.caseListNameSet,
-                this.props.user
-            );
+        this.setDynamicTypeTo(false);
```

Comment:
maybe a more "react" correct way would be to have `dynamic` boolean variable be a state and let the description react to that.

Reply:
@pieterlukasse do you have an example in the code of this approach?

Reply:
i think @pieterlukasse is right.  it seems you want to keep dynamic as a observable state and then make the rest a computation which depends upon it.   you will see the ```@computed``` decorator used thoughout our codebase and that is what it accomplishes.   any getter that is tagged as computed will recompute whenever the observable state which it references changes.

Reply:
@alisman @pieterlukasse Please check https://github.com/cBioPortal/cbioportal-frontend/pull/5016/commits/7871b78e41d78ffcff1933573f3d459f06860d97